src/triage/component/catwalk/utils.py

```diff
@@ -149,20 +150,37 @@ def __iter__(self):
             yield self.group()
 
 
-def sort_predictions_and_labels(predictions_proba, labels, sort_seed):
+def sort_predictions_and_labels(predictions_proba, labels, tiebreaker='random', sort_seed=None):
+    """Sort predictions and labels with a configured tiebreaking rule
+
+    Args:
+        predictions_proba (numpy.array) The predicted scores
+        labels (numpy.array) The numeric labels (1/0, not True/False)
+        tiebreaker (string) The tiebreaking method ('best', 'worst', 'random')
+        sort_seed (signed int) The sort seed. Needed if 'random' tiebreaking is picked.
+
+    Returns:
+        (tuple) (predictions_proba, labels), sorted
+    """
     if len(labels) == 0:
         logging.debug("No labels present, skipping sorting.")
         return predictions_proba, labels
-    else:
+    if tiebreaker == 'random':
+        if not sort_seed:
+            raise ValueError("If random tiebreaker is used, a sort seed must be given")
         random.seed(sort_seed)
-        predictions_proba_sorted, labels_sorted = zip(
-            *sorted(
-                zip(predictions_proba, labels),
-                key=lambda pair: (pair[0], random.random()),
-                reverse=True,
-            )
-        )
-        return predictions_proba_sorted, labels_sorted
+        numpy.random.seed(abs(sort_seed))
```

Comment:
Do we want to disallow negative random seeds rather than alter them silently?

Reply:
Sure. Python allows them but we could just generate them between 0 and whatever instead of -whatever/2 and +whatever/2, so no conversion to make them numpy-able would be needed